lib/ramble/ramble/application.py

```diff
@@ -650,13 +656,84 @@ def _get_executables(self):
             executables = self.internals[namespace.executables]
 
         # Define custom executables
-        if namespace.custom_executables in self.internals.keys():
+        if namespace.custom_executables in self.internals:
             for name, conf in self.internals[namespace.custom_executables].items():
                 self.executables[name] = ramble.util.executable.CommandExecutable(
                     name=name,
                     **conf
                 )
 
+        # Perform executable injection
+        if namespace.executable_injection in self.internals:
+
+            supported_orders = set(['before', 'after'])
```

Comment:
Should we DRY this out into a keyword/enum? These magic strings are already in a few places and previous experience says they might only spread :P

Reply:
This is actually the only place these magic strings show up. For the chain we use `before` and `after`, but the strings are `before_root` `after_root` etc...

Most other places, we use append and prepend, but those are used to control position relative to an entire list rather than within the list.

Reply:
For that comment specifically I was thinking about the local use here, eg: before/after in

```
 if relative is None:
                    if injection_order == 'before':
                        executables.insert(0, exec_name)
                    elif injection_order == 'after':
                        executables.append(exec_name)
                else:

                    found = False
                    if injection_order == 'before':
```

Long term I was indeed thinking about the stuff you mentioned, but agree its OK for now

Reply:
OK, thanks. I pushed an update. Let me know if that looks better.